### DISCUSSION TITLE
如何导入较大的csv文件或者json文件

### DISCUSSION THREAD

**Discussion by LiJie20190102** [milvus-io/milvus#23440] [Q&A and General discussion] [answered]
我是用Attu可以导入csv文件，但是数据大小有限制。所以我想使用CLI或者java SDK来导入数据，但是都遇到了问题：

1、使用CLI时，官网说可以导入本地csv，但是导入时报错：
![image](https://user-images.githubusercontent.com/53458004/232014691-bd83a1a9-9bad-40c1-b3bd-dba739066d5e.png)
这个错误是不是说他用JSON来解析的，但是当我传入JSON文件时，又说我不是csv文件。可能我的操作有误，所以想问下如何使用CLI导入CSV文件数据。
csv文件格式如下（这个csv文件通过Attu是可以正常导入）：
![image](https://user-images.githubusercontent.com/53458004/232015284-db614695-ef7f-4730-a8dd-42c6e44f6535.png)

2、使用java SDK时，官网说添加file，但是我没找到如何指定file的路径在哪里？我尝试传入绝对路径，但是不好使：
![image](https://user-images.githubusercontent.com/53458004/232015765-a320dca6-9266-4439-b4e4-53255b0b14b9.png)

**Comment by yhmo** [comment, marked as answer]
Attu是一个web server，它帮你把csv文件解析为向量数据，然后调用milvus insert接口插入。
CLI是一个sdk，它直接调用milvus的bulk_insert接口，该接口只支持Json和Numpy文件。

用Java SDK的话，用bulkinsert接口来导入文件：
R<ImportResponse> bulkInsert(BulkInsertParam requestParam);

导入文件的做法和注意事项在文档里： https://milvus.io/docs/bulk_insert.md

**Reply by LiJie20190102** [reply]
我看milvus.yml的minio.bucketName配置是a-bucket
![image](https://user-images.githubusercontent.com/53458004/232071501-aea44eef-f639-4d08-b929-1092cda36d58.png)

我把文件放到minio容器的这个/minio_data/a-bucket目录：
![image](https://user-images.githubusercontent.com/53458004/232071293-52bea3d2-ece1-4e1d-998f-ddcb5389430b.png)
我在java SDK中这样使用时还是有点问题，是我哪里使用有问题么？
![image](https://user-images.githubusercontent.com/53458004/232071947-62bcb630-798a-47f4-846a-92627ade9174.png)

**Reply by LiJie20190102** [reply]
已经使用Java DK解决，我之前minio数据没有导入好

**Reply by sunmingqiaa** [reply]
我遇到跟你一样的问题，请问你是怎么解决的

**Reply by yhmo** [reply]
> 我遇到跟你一样的问题，请问你是怎么解决的

关键点是要把json/numpy文件上传到minio的 a-bucket 之下，不要直接拷贝到容器的/minio_data/a-bucket目录。上传文件的方法，可以用minio的sdk来做，也可以用浏览器打开  localhost:9001在minio的控制台里做。

**Reply by sunmingqiaa** [reply]
非常感谢

**Reply by sunmingqiaa** [reply]
@yhmo 请问python sdk有相关文档吗，我再官网没找到

**Reply by yhmo** [reply]
> @yhmo 请问python sdk有相关文档吗，我再官网没找到

https://milvus.io/docs/bulk_insert.md
